pallets/market/src/lib.rs

```diff
@@ -433,7 +433,11 @@ pub mod pallet {
             let caller = ensure_signed(origin)?;
 
             BalanceTable::<T>::try_mutate(&caller, |balance| -> DispatchResult {
-                ensure!(balance.free >= amount, Error::<T>::InsufficientFreeFunds);
+                if balance.free < amount {
+                    log::error!(target: LOG_TARGET, "withdraw_balance: not enough free balance {:?} < {:?}", balance.free, amount);
+                    fail!(Error::<T>::InsufficientFreeFunds);
+                }
+
```

Comment:
Instead you could do the other hack:
```suggestion
				ensure!(balance.free >= amount, {
					log::error!(target: LOG_TARGET, "withdraw_balance: not enough free balance {:?} < {:?}", balance.free, amount);
					Error::<T>::InsufficientFreeFunds
				});
```